Python/Product/PythonTools/PythonTools/Repl/PythonDebugReplEvaluator.cs

```diff
@@ -421,15 +410,38 @@ internal class PythonDebugReplEvaluator :
                 return;
             }
 
+            // Multiple callers can execute in this method simultaneously (due to async/await)
+            // so we need to protect against 2 callers trying to create evaulators for the same process.
+
+            // Keep track of evaluators that are in progress of attaching, and if
+            // we are getting called to attach for one that is already in progress,
+            // just wait for it to finish before returning.
+            Task attachingTask;
+            TaskCompletionSource<object> attachingTcs = null;
+            if (_attachingTasks.TryGetValue(process.Id, out attachingTask)) {
```

Comment:
I'll try to clarify the comment above. Locking isn't needed because the simultaneous callers are both on the UI thread. The second UI thread caller won't enter this function until the first caller uses await down below and yields execution. So as long as we check the dictionary before we call await, it's ok because they aren't really running in parallel.